google-cloud-spanner/src/main/java/com/google/cloud/spanner/SingleDmlTransaction.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import com.google.cloud.spanner.Options.TransactionOption;
+import com.google.cloud.spanner.Options.UpdateOption;
+import com.google.cloud.spanner.SessionPool.PooledSessionFuture;
+import com.google.common.annotations.VisibleForTesting;
+import io.opencensus.common.Scope;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import java.util.Arrays;
+
+public class SingleDmlTransaction {
+  private static final String READ_WRITE_TRANSACTION = "CloudSpanner.ReadWriteTransaction";
+  private static final Tracer tracer = Tracing.getTracer();
+  private final TransactionRunner runner;
+
+  SingleDmlTransaction(PooledSessionFuture session, TransactionOption... options) {
+    Span span = tracer.spanBuilder(READ_WRITE_TRANSACTION).startSpan();
+    try (Scope s = tracer.withSpan(span)) {
+      TransactionOption[] new_options = Arrays.copyOf(options, options.length + 1);
+      new_options[options.length] = Options.INLINE_COMMIT_OPTION;
+      runner = session.readWriteTransaction(new_options);
+    } catch (RuntimeException e) {
+      TraceUtil.endSpanWithFailure(span, e);
+      throw e;
+    } finally {
+      span.end(TraceUtil.END_SPAN_OPTIONS);
+    }
+  }
+
+  @VisibleForTesting
+  public SingleDmlTransaction(TransactionRunner runner) {
+    this.runner = runner;
+  }
+
+  public long executeUpdate(Statement statement, UpdateOption... options)
+      throws NullPointerException, SpannerException {
```

Comment:
nit: no need to put the exceptions in the function signature, since they are unchecked exceptions

Reply:
removed